.github/scripts/moss.pl

```diff
@@ -0,0 +1,354 @@
+# large number), moss reports all matches, no matter how often they appear.  
+# The -m setting is most useful for large assignments where one also a base file 
+# expected to hold all legitimately shared code.  The default for -m is 10.
+#
+# Examples:
+#
+#   moss -l pascal -m 2 *.pascal 
+#   moss -l cc -m 1000000 -b mycode.cc asn1/*.cc
+#
+# 
+# The -c option supplies a comment string that is attached to the generated
+# report.  This option facilitates matching queries submitted with replies
+# received, especially when several queries are submitted at once.
+#
+# Example:
+#
+#   moss -l scheme -c "Scheme programs" *.sch
+#
+# The -n option determines the number of matching files to show in the results.
+# The default is 250.
+#
+# Example:
+#   moss -c java -n 200 *.java
+# The -x option sends queries to the current experimental version of the server.
+# The experimental server has the most recent Moss features and is also usually
+# less stable (read: may have more bugs).
+#
+# Example:
+#
+#   moss -x -l ml *.ml
+#
+
+
+#
+# Section 2.  Installation instructions.
+#     
+# You may need to change the very first line of this script
+# if perl is not in /usr/bin on your system.  Just replace /usr/bin
+# with the pathname of the directory where perl resides.
+#
+
+#
+#  3. Standard Copyright
+#
+#Copyright (c) 1997 The Regents of the University of California.
+#All rights reserved.
+#
+#Permission to use, copy, modify, and distribute this software for any
+#purpose, without fee, and without written agreement is hereby granted,
+#provided that the above copyright notice and the following two
+#paragraphs appear in all copies of this software.
+#
+#IN NO EVENT SHALL THE UNIVERSITY OF CALIFORNIA BE LIABLE TO ANY PARTY FOR
+#DIRECT, INDIRECT, SPECIAL, INCIDENTAL, OR CONSEQUENTIAL DAMAGES ARISING OUT
+#OF THE USE OF THIS SOFTWARE AND ITS DOCUMENTATION, EVEN IF THE UNIVERSITY OF
+#CALIFORNIA HAS BEEN ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#
+#THE UNIVERSITY OF CALIFORNIA SPECIFICALLY DISCLAIMS ANY WARRANTIES,
+#INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY
+#AND FITNESS FOR A PARTICULAR PURPOSE.  THE SOFTWARE PROVIDED HEREUNDER IS
+#ON AN "AS IS" BASIS, AND THE UNIVERSITY OF CALIFORNIA HAS NO OBLIGATION TO
+#PROVIDE MAINTENANCE, SUPPORT, UPDATES, ENHANCEMENTS, OR MODIFICATIONS.
+#
+#
+# STOP.  It should not be necessary to change anything below this line
+# to use the script.
+#
+use IO::Socket;
+
+#
+# As of the date this script was written, the following languages were supported.  This script will work with 
+# languages added later however.  Check the moss website for the full list of supported languages.
+#
+@languages = ("c", "cc", "java", "ml", "pascal", "ada", "lisp", "scheme", "haskell", "fortran", "ascii", "vhdl", "perl", "matlab", "python", "mips", "prolog", "spice", "vb", "csharp", "modula2", "a8086", "javascript", "plsql");
+
+$server = 'moss.stanford.edu';
```

Comment:
uhhh is this like, making a bunch of api calls to some stanford server....?

Reply:
Yea, we use the code from http://moss.stanford.edu/general/scripts/mossnet

Reply:
The moss.pl script is the client-side utility provided by the MOSS service. This script helps to send our game files to the MOSS server at Stanford University for analysis.

Reply:
this doesn't seem like it's set up for production workloads...  this is very unusual/weird...

Reply:
heh for instance starting w/ this: 
![Screenshot 2024-09-16 at 11 31 07 AM](https://github.com/user-attachments/assets/79cc6644-ee6e-4777-8c6e-beee020545e0)